pyquil/quil.py

```diff
@@ -377,33 +378,7 @@ def _add_instruction(self, instruction: quil_rs.Instruction) -> None:
         For backwards compatibility, it also prevents duplicate calibration, measurement, and gate definitions from
         being added. Users of ``Program`` should use ``inst`` or ``Program`` addition instead.
         """
-        if instruction.is_gate_definition():
```

Comment:
Since `quil-py` stores DEFGATEs in a map keyed on name they are automatically de-duped, making this branch redundant.